### PR TITLE
Cleanup, updates and made more generic

### PR DESCRIPTION
**Pull request by Superjo149**
This is an improvement to the original module to make it more extensible for other users like myself. I think this is a good basis to build upon further if needed.

Instead of the existing class, I created 2 separate classes. One with the playlist, and one for playing a single track. Please see the README or code for more information.

### CODE REVIEW

**Comment by Superjo149** [comment]
@imsingh Are you able to take a look at this? Or should I make a separate package from it, as there are some API changes?

**Comment by imsingh** [comment]
Hi there. Thanks for the pull request. I'll take a look over weekend.

File: README.md [hunk 1]
```diff
@@ -30,79 +30,59 @@ yarn install rxjs-audio
...
+### Initialize the stream
+
+Before using, you need to create an object of the Stream class. You'll need to pick a class depening on your use-case: 
+- `AudioStream` is the base-class similar to the html5 audio element where you can load a track. 
+- `PlaylistAudioStream` is like the name says more suitable for playing playlists, in the options you'll be able to add an array of tracks
 
-### Creating a Stream
 ```ts
-import { RxJSAudioService } from 'rxjs-audio';
+import { AudioStream } from 'rxjs-audio-stream';
```
Comment (imsingh) [inline README.md:40]: Can you please rename it back to the original name of the package? or there is a specific reason you did it?

**Comment by imsingh** [comment]
@Superjo149 I like the idea. I'd be happy to merge this. Can you create a demo app also so that I do a final check?
I'm also okay if creating a new package helps you with your project.

File: README.md (hunk 1 above)
Comment (Superjo149) [reply README.md:40]: My bad, will update this

**Comment by Superjo149** [comment]
@imsingh I am already using it here, https://github.com/Superjo149/auryo/blob/392e94ebebb13670ddc96ff13f90739dc71f6490/src/renderer/epics/audio.ts#L31> But the branch itself is very much a WIP, so you will not be able to easily test it. I can maybe see to create a small dummy thing for it if you want.

**Comment by Superjo149** [comment]
I have added a super simple example to illustrate

**Comment by imsingh** [comment]
@Superjo149 Hi. I understand you want it to be merged. It's likewise. I had pretty rough days at my work. This weekend, I'll merge it, update the companion app and release a new version. Just a little longer.

Thanks for your patience and work!

**Comment by Superjo149** [comment]
@imsingh Allright, thanks. No worries, it was just a friendly reminder 🙂

**Review by imsingh — APPROVED** [review]

**Pull request merged**